### PR TITLE
Sprint 34

### PR DESCRIPTION
**Pull request by ericsorenson**
# Sprint 34

### CODE REVIEW

**Comment by waldoj** [comment]
### Code Review Checklist

We use this list when performing a code review to ensure that all tasks have been completed.

- [ ] begin by deleting any existing copy of the Git repository, running `docker system prune` to remove any cached artifacts from any prior code review
- [ ] clone a fresh copy of the Git repository (`git clone https://github.com/ustaxcourt/ef-cms.git`), fetch the pull request and name it for the sprint (e.g., `git fetch origin pull/204/head:sprint-43`), and then switch to that branch (e.g. `git checkout sprint-43`)
- [ ] review the pull request itself, to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] stand up the site locally, with `./docker-run.sh`
	- [ ] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [ ] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [ ] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [ ] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [ ] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [ ] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [ ] review for accessibility
		- [ ] use an automated audit tool, such as Chrome Audit or aXe
		- [ ] navigate site only with the keyboard
		- [ ] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [ ] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [ ] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [ ] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [ ] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [ ] methodically review all new tests for correctness, quality of naming
- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [ ] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

File: web-client/terraform/common/cloudfront-edge/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    requestHeaders['x-allowed-domain'][0].value
+  ) {
+    allowedDomain = requestHeaders['x-allowed-domain'][0].value;
+  }
+  const allowedDomainString = allowedDomain ? `*.${allowedDomain}` : '';
+
+  //Set new headers
+  headers['strict-transport-security'] = [
+    {
+      key: 'Strict-Transport-Security',
```
Comment (waldoj) [inline web-client/terraform/common/cloudfront-edge/index.js]: 👏🏻

File: web-client/terraform/common/cloudfront-edge/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    `form-action ${applicationUrl}`,
+    "object-src 'none'",
+    `script-src 'self' ${dynamsoftUrl} 'unsafe-inline' resource://pdf.js`,
+    `style-src 'self' 'unsafe-inline' ${dynamsoftUrl}`,
+  ];
+  headers['content-security-policy'] = [
+    {
+      key: 'Content-Security-Policy',
+      value: contentSecurityPolicy.join('; '),
+    },
```
Comment (waldoj) [inline web-client/terraform/common/cloudfront-edge/index.js]: 👏🏻

File: web-client/terraform/common/cloudfront-edge/index.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  headers['x-content-type-options'] = [
+    { key: 'X-Content-Type-Options', value: 'nosniff' },
+  ];
+  headers['x-frame-options'] = [{ key: 'X-Frame-Options', value: 'DENY' }];
+  headers['x-xss-protection'] = [
+    { key: 'X-XSS-Protection', value: '1; mode=block' },
+  ];
+  headers['referrer-policy'] = [
+    { key: 'Referrer-Policy', value: 'same-origin' },
+  ];
```
Comment (waldoj) [inline web-client/terraform/common/cloudfront-edge/index.js]: This is all great. :)

File: shared/src/business/useCaseHelper/standingPretrialNotice/generateStandingPretrialNoticeTemplate.test.js [hunk 4]
```diff
@@ -37,10 +45,49 @@ describe('generateStandingPretrialNoticeTemplate', () => {
...
-    //expect(result.indexOf('ST')).toBeGreaterThan(-1);
+    expect(result.indexOf('Test Case Caption')).toBeGreaterThan(-1);
+    expect(result.indexOf('123-45S')).toBeGreaterThan(-1);
+    expect(result.indexOf('City')).toBeGreaterThan(-1);
+    expect(result.indexOf('ST')).toBeGreaterThan(-1);
+    expect(result.indexOf('Guy Fieri')).toBeGreaterThan(-1);
+    expect(result.indexOf('STANDING PRETRIAL NOTICE')).toBeGreaterThan(-1);
+  });
+
+  it('Returns HTML with the given case and trial session data with an incorrect respondent structure', async () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply shared/src/business/useCaseHelper/standingPretrialNotice/generateStandingPretrialNoticeTemplate.test.js]: I believe the intention was to test for 100% branch coverage. In the code we have:
```
if (trialInfo.respondents && trialInfo.respondents.length) { ... }
```

Providing an object instead of an array will hit the `trialInfo.respondents.length` evaluation to give us that branch coverage. I'm updating the description and adding a comment that points this out in the test data.

[the remark this answers is not in the document]
Comment (matthopson) [reply shared/src/business/useCaseHelper/standingPretrialNotice/generateStandingPretrialNoticeTemplate.test.js]: I'm also going to remove those //Todo  comments and flesh the test out a little more while I'm in there :)

File: web-client/src/presenter/actions/WorkItem/assignPetitionToAuthenticatedUserAction.test.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    await runAction(assignPetitionToAuthenticatedUserAction, {
+      modules: {
+        presenter,
+      },
+    });
+
+    expect(assignWorkItemsInteractorStub).not.toHaveBeenCalled();
+  });
+
+  it('should assign the workitem if the qc work item is present', async () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply web-client/src/presenter/actions/WorkItem/assignPetitionToAuthenticatedUserAction.test.js]: Ah, there's a typo. `docuementType` -> `documentType` gets the test to fail, so we can fix the expect statement to match what we _really_ expect. It _should_ get assigned.

File: web-client/src/presenter/actions/caseDetailEdit/setCaseCaptionForCaseInfoTabAction.test.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { runAction } from 'cerebral/test';
+import { setCaseCaptionForCaseInfoTabAction } from './setCaseCaptionForCaseInfoTabAction';
+
+describe('setCaseCaptionForCaseInfoTabAction', () => {
+  it('should set state.form.caseCaption to the passed in props.caseCaption if the tab is caseInfo', async () => {
```
[the remark this answers is not in the document]
Comment (matthopson) [reply web-client/src/presenter/actions/caseDetailEdit/setCaseCaptionForCaseInfoTabAction.test.js]: Test description is wrong. It should say `partyInfo` - the relevant code:
```
if (tab === 'partyInfo') {
    store.set(state.caseDetail.caseCaption, props.caseCaption);
  }
```

**Comment by waldoj** [comment]
@JessicaMarine, I have reviewed this pull request to `staging`, I am satisfied with the changes that Flexion has made, and I recommend that it be merged.
